phpdude/django-macros-url
# Add uuid version to uuid regex

### What is the problem / feature ?

The regex for uuid's didn't enforce the `version` part of the uuid.  

```
4.1.3.  Version

   The version number is in the most significant 4 bits of the time
   stamp (bits 4 through 7 of the time_hi_and_version field).

   The following table lists the currently-defined versions for this
   UUID variant.

   Msb0  Msb1  Msb2  Msb3   Version  Description

    0     0     0     1        1     The time-based version
                                     specified in this document.

    0     0     1     0        2     DCE Security version, with
                                     embedded POSIX UIDs.
```

The first digit in the 3rd chunk of the uuid indicates the version.  It can be `1, 2, 3, or 5`.

### How did it get fixed / implemented ?

Adjusted the regex to reflect this restriction.

### How can someone test / see it ?

Try to match something like `'3e41b04d-0978-9027-86c2-aa90c63ecb54'` and see it fails.

*Here is a cute animal picture for your troubles...*

![1411729104079_wps_39_pic_from_caters_news_pict](https://cloud.githubusercontent.com/assets/824194/6622073/ea7063f2-c89f-11e4-89b4-87c11f64f269.jpg)
